### ISSUE TITLE
Potential Nuxt.js Error: cannot enable cancellation after promises are in use

### ISSUE THREAD

**Issue by cprater** [AlbertLucianto/vuex-search#8]
Hey @AlbertLucianto I'm having a difficult time getting this plugin to work after installing and setting up a basic `searchPlugin()`. I'm using vuex-search in a [Nuxt.js](https://nuxtjs.org/) environment and have done the following:

- `yarn add vuex-search` which installed `v2.2.2`
- In my `store/index.js`:
```
export const state = () => ({
  'pages': []
});

export const plugins = [
  searchPlugin({
    'resources': {
      'pages': {
        'index': ['title'],
        'getter': state => state.pages,
        'watch': { delay: 500 },
      }
    }
  })
];
```

- But when I run `yarn start` to start the application, I get the error with the following stack trace:
```
 Error: cannot enable cancellation after promises are in use
 
 - debuggability.js:282 Function.Promise.config
   [project-name]/[bluebird]/js/release/debuggability.js:282:19
 
 - utils.js:85 cancellablePromiseWrapper
   [project-name]/[vuex-search]/dist/commonjs/utils.js:85:22
 
 - SearchApi.js:173 SubscribableSearchApi._callee$
   [project-name]/[vuex-search]/dist/commonjs/SearchApi.js:173:70
 
 - runtime.js:62 tryCatch
   [project-name]/[regenerator-runtime]/runtime.js:62:40
 
 - runtime.js:296 Generator.invoke [as _invoke]
   [project-name]/[regenerator-runtime]/runtime.js:296:22
 
 - runtime.js:114 Generator.prototype.(anonymous function) [as next]
   [project-name]/[regenerator-runtime]/runtime.js:114:21
 
 - asyncToGenerator.js:17 step
   [project-name]/[babel-runtime]/helpers/asyncToGenerator.js:17:30
 
 - asyncToGenerator.js:35 
   [project-name]/[babel-runtime]/helpers/asyncToGenerator.js:35:14
 
 - new Promise
 
 - _export.js:36 new F
   [project-name]/[core-js]/library/modules/_export.js:36:28
 
 - asyncToGenerator.js:14 SubscribableSearchApi.<anonymous>
   [project-name]/[babel-runtime]/helpers/asyncToGenerator.js:14:12
 
 - SearchApi.js:218 SubscribableSearchApi.performSearch
   [project-name]/[vuex-search]/dist/commonjs/SearchApi.js:218:22
 
 - actions.js:53 Store.<anonymous>
   [project-name]/[vuex-search]/dist/commonjs/actions.js:53:29
 
 - vuex.common.js:706 Array.wrappedActionHandler
   [project-name]/[vuex]/dist/vuex.common.js:706:23
 
 - vuex.common.js:428 Store.dispatch
   [project-name]/[vuex]/dist/vuex.common.js:428:15
 
 - vuex.common.js:334 Store.boundDispatch [as dispatch]
   [project-name]/[vuex]/dist/vuex.common.js:334:21
```

I don't understand why the error is coming from the plugin. This makes me feel like I've missed some very obvious setup step, any ideas?

**Comment by AlbertLucianto** [comment]
@cprater Your code looks fine. It seems the problem is related to one of vuex-search dependencies, which is bluebird for cancelling outdated search.

If that really is the case, probably I will replace bluebird with a custom handler for promise cancellation in the next version.

**Comment by AlbertLucianto** [comment]
Just deployed and should be fixed in v2.2.3.
